src/amo/reducers/reviews.js

```diff
@@ -576,6 +579,12 @@ export default function reviewsReducer(
       }
       return newState;
     }
+    case SET_VIEW_CONTEXT: {
+      return {
+        ...state,
+        view: {},
```

Comment:
This is brutal because it resets all view state for reviews / ratings, some of which may be used in unrelated components.

All we need to do is hide the form so I suggest `editingReview: false`. If we need to reset more things in the future we could but for now I think that's all we need.

By the way, I considered just hiding the form explicitly on unmount. This works but it breaks hot reloading so I don't think it's a good solution.

In `AddonReviewManager`, I added:

```js
componentWillUnmount() {
  const { review, dispatch } = this.props;
  dispatch(hideEditReviewForm({ reviewId: review.id }));
}
```

Reply:
Oh, I see how there's a problem with not having `reviewId`. 

I haven't looked into the implications of resetting the entire view state but if you think it's harmless then I suppose it will work.

The only problem with `componentWillUnmount()` is how it breaks hot reloading. That's not a deal breaker, just an annoyance. If that's a better way to solve it then it's fine.

Reply:
I've looked through all of the code that uses the `reviews.view` state, and it seems to me that all of it is used for maintaining state on an individual page. I think if the location changes, that would render all of that data useless/unnecessary, so I do think it would be fine to clear the whole thing out on a `LOCATION_CHANGE` action. This will also avoid breaking hot reloading. I have made this change.